pc/PlatformDeviceControl.c

```diff
@@ -35,9 +35,14 @@ int usbFdWrite = -1;
 int usbFdRead = -1;
 #endif  /*USE_USB_VSC*/
 
+#include "XLinkPublicDefines.h"
+
 #define USB_LINK_SOCKET_PORT 5678
 #define UNUSED __attribute__((unused))
 
+
+static UsbSpeed_t usb_speed_enum = X_LINK_USB_SPEED_UNKNOWN;
```

Comment:
these global variables are not thread safe

Reply:
and on threadsafe I am actually copying it (here)[https://github.com/luxonis/XLink/blob/53da7b26ecbf4bcc617802c26464fc3ffb97efce/shared/src/XLinkDevice.c#L215] so I just let it be. but yeah you are right it should be threadsafe. I will check on that.

Reply:
Seems like `XLinkConnect` isn't threadsafe, and both write and read to those variables happen in that function call, so I think that is okay from that standpoint

Reply:
`static char mx_serial[128] = { 0 };` 
can rather be
`static char mx_serial[XLINK_MAX_MXID] = { 0 };`

Reply:
I modified [here](https://github.com/luxonis/depthai-shared/blob/93ac1e512b38043ae97f99f5381035bc780896e4/src/xlink/xlink_wrapper.cpp#L357) in xlink wrapper to make it threadsafe.